src/commands/moderation/prefixCommands/functions/addVersion.ts

```diff
@@ -0,0 +1,142 @@
+    color: Colors.Green,
+});
+
+const modLogEmbed = (moderator: User, version: string, emoji: string, alias: string, enabled: boolean, versionId: string) => makeEmbed({
+    title: 'Prefix command version added',
+    fields: [
+        {
+            name: 'Version',
+            value: version,
+        },
+        {
+            name: 'Moderator',
+            value: `${moderator}`,
+        },
+        {
+            name: 'Emoji',
+            value: emoji,
+        },
+        {
+            name: 'Alias',
+            value: alias,
+        },
+        {
+            name: 'Enabled',
+            value: enabled ? 'Yes' : 'No',
+        },
+    ],
+    footer: { text: `Version ID: ${versionId}` },
+    color: Colors.Green,
+});
+
+const noModLogs = makeEmbed({
+    title: 'Prefix Commands - Add Version - No Mod Log',
+    description: 'I can\'t find the mod logs channel. Please check the channel still exists.',
+    color: Colors.Red,
+});
+
+export async function handleAddPrefixCommandVersion(interaction: ChatInputCommandInteraction<'cached'>) {
+    await interaction.deferReply({ ephemeral: true });
+
+    const conn = getConn();
+    if (!conn) {
+        await interaction.followUp({ embeds: [noConnEmbed], ephemeral: true });
+        return;
+    }
+
+    const name = interaction.options.getString('name')!;
+    const emoji = interaction.options.getString('emoji')!;
+    const alias = interaction.options.getString('alias')!.toLowerCase();
+    const enabled = interaction.options.getBoolean('is_enabled') || false;
+    const moderator = interaction.user;
+
+    const nameRegex = /^[\w-]+$/;
+    if (!nameRegex.test(name)) {
+        await interaction.followUp({ embeds: [wrongFormatEmbed(name)], ephemeral: true });
+        return;
+    }
+    if (!nameRegex.test(alias)) {
+        // eslint-disable-next-line no-await-in-loop
+        await interaction.followUp({ embeds: [wrongFormatEmbed(alias)], ephemeral: true });
+        return;
+    }
+
+    // Check if a command or version exists with the same name or alias
+    const foundCommandName = await PrefixCommand.findOne({
+        $or: [
+            { name: alias },
+            { aliases: alias },
+        ],
+    });
+    if (foundCommandName) {
+        await interaction.followUp({ embeds: [alreadyExistsEmbed(name, `${alias} already exists as a command or alias.`)], ephemeral: true });
+        return;
+    }
+    const foundVersion = await PrefixCommandVersion.findOne({
+        $or: [
+            { name },
+            { alias },
+        ],
+    });
+    if (foundVersion || name.toLowerCase() === 'generic' || alias === 'generic') {
+        await interaction.followUp({ embeds: [alreadyExistsEmbed(name, `${alias} already exists as a version alias.`)], ephemeral: true });
+        return;
+    }
+
+    //Check if the mod logs channel exists
+    const modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS) as TextChannel;
+    if (!modLogsChannel) {
```

Comment:
```suggestion
    let modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS);
    if (!modLogsChannel || !modLogsChannel.isTextBased()) {
        modLogsChannel = null;
```
Might be safer to actually check if the channel is a text channel? Especially if we want to offer the codebase to other projects. Then, if it's not a text channel set it to null to prevent accidental attempts to send into a non-text channel.